domain/upgrade/service/service_test.go

```diff
@@ -168,7 +168,7 @@ func (s *serviceSuite) TestWatchForUpgradeState(c *gc.C) {
 
 	nw := watchertest.NewMockNotifyWatcher(nil)
 
-	s.watcherFactory.EXPECT().NewValueMapperWatcher(gomock.Any(), s.upgradeUUID.String(), gomock.Any(), gomock.Any()).Return(nw, nil)
+	s.watcherFactory.EXPECT().NewNotifyMapperWatcher(gomock.Any(), gomock.Any()).Return(nw, nil)
```

Comment:
We should be checking the filter arg, otherwise we're missing test coverage. Case in point is the line deleted which checks that the expected upgrade UUID is used. Given the parameter is a FilterOption interface, can we implement a go mock matcher to use.
This comment would apply elsewhere if the same thing is done.